### PR TITLE
Feature: Docker

### PR DESCRIPTION
**Pull request by seabbs**
This PR adds: 

- A docker container
- Included and updated dockerfile
- vscode setup to run out the box using the remote-development plugin
- A github action to build the dockerfile
- Adds MIT license
- Adds .gitignore

Dockerfile has been tested to build but not that it runs the code (as not yet present!). I have a clean build setup to test the code as it comes in.

### CODE REVIEW

File: .devcontainer/library-scripts/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# install precommit
+RUN pip install pre-commit
+
+# add docthis
+RUN Rscript -e 'devtools::install_github("mdlincoln/docthis")'
+
+# install precommit
+RUN Rscript -e 'devtools::install_github("lorenzwalthert/precommit")'
+
+RUN Rscript -e 'devtools::install_github("stan-dev/cmdstanr")'
```
Comment (sbfnk) [inline .devcontainer/library-scripts/Dockerfile]: Are these all needed (git-lfs/docthis/precommit/cmdstanr)?

File: DESCRIPTION [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    person("Nikos", "Bosse", , "nikosbosse@gmail.com", role = "aut"),
+    person("Katharine", "Sherratt", , "katharine.sherratt@lshtm.ac.uk", role = "aut",
+           comment = c(ORCID = "0000-0003-4413-453X")),
+    person("Sebastian", "Funk", , "sebastian.funk@lshtm.ac.uk", role = "aut")
+  )
+Description: Nowcasts and Projections for Covid-19 in the UK
+License: MIT + file LICENSE
+Depends: 
+    R (>= 3.4.0)
+Imports:
```
Comment (sbfnk) [inline DESCRIPTION:19]: I suspect we may not need these all - keeping a running list.

File: DESCRIPTION [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    patchwork,
+    purrr,
+    readr,
+    rmarkdown,
+    stringr,
+    stringr,
+    tibble,
+    tidyr
+Remotes:
+    epiforecasts/EpiNow2@ca818edfe80c05f873ef9255e12158e2ce60430d
```
Comment (sbfnk) [inline DESCRIPTION]: any reason not to pull in latest?

File: .devcontainer/library-scripts/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.205.2/containers/r
+{
+	"name": "ct-modelling",
```
Comment (sbfnk) [inline .devcontainer/library-scripts/devcontainer.json]: rename?

**Review by sbfnk — APPROVED** [review]

File: .devcontainer/library-scripts/Dockerfile (hunk 1 above)
Comment (seabbs) [reply .devcontainer/library-scripts/Dockerfile]: None of them are (and I thought I had already removed. )

**Comment by seabbs** [comment]
> any reason not to pull in latest? 

Just in case there is an interface change that breaks the code. Depends if better to get potential bug fixes or not I suppose.

**Comment by seabbs** [comment]
> rename? 

Ah damn this is missing a commit

File: DESCRIPTION [hunk 5]
```diff
@@ -0,0 +1,47 @@
+Type: Package
+Package: Covid19UKNowcastsProjections
```
Comment (sbfnk) [inline DESCRIPTION:2]: I hope I'll never have to type this because there's no way I'd get the capitalisation right ;-)

Comment (seabbs) [reply DESCRIPTION:2]: Annoyingly won't allow hyphens so can't match the repo name.

**Pull request merged**